src/FactoryTrait.php

```diff
@@ -133,11 +136,17 @@ public function factory($seed, $defaults = [], string $prefix = null): object
             return $seed;
         }
 
-        $object = array_shift($seed); // first numeric key argument is object
-
         $arguments = array_filter($seed, 'is_numeric', ARRAY_FILTER_USE_KEY); // with numeric keys
         $injection = array_diff_key($seed, $arguments); // with string keys
 
+        $object = array_shift($arguments); // first numeric key argument is object
+        if (!$object) {
+            throw new Exception([
+                'factory() could not find object',
+                'seed' => $seed,
+            ]);
+        }
+
         if (is_string($object)) {
```

Comment:
@DarkSide666 if we put `!is_object($object)` here, would it work? We can then remove the if condition above, it is bad, as it emits about the same meaning with completely different message.

Reply:
will `normalizeClassName(null, $prefix)` not throw exception then?
it will because `normalizeClassName()` only accepts 1st parameter as string type and nothing more.